distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1966,14 +1952,105 @@ private CometDClient setupCometDClient(List<String> channels) throws Exception {
         cometDClient.start();
-        channels.stream()
-                .forEach(cometDClient::subscribe);
+        channels.forEach(cometDClient::subscribe);
         return cometDClient;
     }
 
+    private void setupParallelInterruptDownloadTest(String metacardId, String filename,
+            String resourceData) throws Exception {
+        cswServer.whenHttp()
+                .match(post("/services/csw"),
+                        withPostBodyContaining("GetRecords"),
+                        withPostBodyContaining(metacardId))
+                .then(ok(),
+                        contentType("text/xml"),
+                        bytesContent(getCswQueryResponse(metacardId).getBytes()));
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.parameter("id", metacardId))
+                .then(getCswRetrievalHeaders(filename), new ChunkedContent.ChunkedContentBuilder(
+                        resourceData).delayBetweenChunks(Duration.ofMillis(50))
+                        .build());
+    }
+
+    /**
+     * Default settings.
+     */
+    private void runParallelInterruptDownloadTest(ClientDownloader client1,
+            ClientDownloader client2) throws Exception {
+        runParallelInterruptDownloadTest(client1,
+                client2,
+                DEFAULT_PARALLEL_CLIENT_START_DELAY,
+                false);
+    }
+
+    /**
+     * Default settings with choice of start order flag.
+     */
+    private void runParallelInterruptDownloadTest(ClientDownloader client1,
+            ClientDownloader client2, boolean flipStartOrder) throws Exception {
+        runParallelInterruptDownloadTest(client1,
+                client2,
+                DEFAULT_PARALLEL_CLIENT_START_DELAY,
+                flipStartOrder);
+    }
+
+    /**
+     * EXECUTION LOGIC FOR PARALLEL INTERRUPTION OF DOWNLOADS
+     * In this template:
+     * We always start client2 first, unless the flipStartOrder flag is set
+     * We always interrupt client1
+     * We always validate the output integrity of client2
+     */
+    private void runParallelInterruptDownloadTest(ClientDownloader client1,
+            ClientDownloader client2, long startDelay, boolean flipStartOrder) throws Exception {
+        String filename = "product1.txt";
+        String metacardId = generateUniqueMetacardId();
+        String resourceData = getResourceData(metacardId);
+
+        setupParallelInterruptDownloadTest(metacardId, filename, resourceData);
+
+        if (flipStartOrder) {
+            client1.startDownload(metacardId);
+            Thread.sleep(startDelay);
+            client2.startDownload(metacardId);
+        } else {
+            client2.startDownload(metacardId);
+            Thread.sleep(startDelay);
+            client1.startDownload(metacardId);
+        }
+
+        Thread.sleep(600);
```

Comment:
This was discussed prior to implementation. It's not the _best_ way to do it but priority was placed on getting tests done so that standard framework code can be safely modified ASAP.

Take a look at `ChunkedContent.java` and you'll see we have facilities in place for slowing down transfer speeds so that download tests consistently meet the criteria. If we are going to make a swap to `WaitCondition` that is a major change that would have to go on a separate ticket. It would require changing the entire infrastructure of download tests.